app/src/main/java/ch/epfl/qedit/view/util/ListEditView.java

```diff
@@ -0,0 +1,211 @@
+package ch.epfl.qedit.view.util;
+
+import static androidx.recyclerview.widget.ItemTouchHelper.ACTION_STATE_DRAG;
+import static androidx.recyclerview.widget.ItemTouchHelper.ACTION_STATE_IDLE;
+import static androidx.recyclerview.widget.ItemTouchHelper.DOWN;
+import static androidx.recyclerview.widget.ItemTouchHelper.UP;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import ch.epfl.qedit.R;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class is a utility class that allows to create editable lists. It supports the addition,
+ * removal and reordering of items. Items additionally have an edit button that can be linked to a
+ * custom action.
+ */
+public class ListEditView extends RecyclerView {
+
+    /**
+     * This interface is used to retrieve the text for one of the items in the list.
+     *
+     * @param <T> the type of the list items
+     */
+    public interface GetItemText<T> {
```

Comment:
I'm a little bit confused about this interface. Either I forgot or I never knew that one could use Interfaces in such a way ... What class is actually going to implement this interface and what does it mean that the interface is defined in the scope of the ListEditView class? This more like a question than request of change to the code.

Reply:
I created that class in order to make the list edit view more generic. That way, you can put a list of an arbitrary type in your list as long as you provide a way to transform an item to its textual representation. That's exactly what this interface is meant for. Note that if we were using a functional language, this would be a lambda, so we wouldn't need all that boilerplate code.

Reply:
And an additional remark: if we were using Java 8, I would have probably used one of the standard functional interface such as `Function<T, String>`. But in Java 7 we have to create those interfaces manually...